tools/c7n_gcp/c7n_gcp/query.py

```diff
@@ -161,13 +167,57 @@ def resources(self, query=None):
                     local_session(self.session_factory).get_default_project())
                 return []
             raise
-        self._cache.save(key, resources)
-        return self.filter_resources(resources)
 
     def augment(self, resources):
         return resources
 
 
+class ChildResourceManager(QueryResourceManager):
+
+    @classmethod
+    def get_parent_annotation_key(cls):
```

Comment:
Create a subclass of type info for child type info, and move this function there. As is this has a wierd bi directional calling syntax, with the type info querying the resource manager by name from within the enclosure, its just static metadata and should be on type info.

Reply:
Agree. Thanks